did_doc/src/schema/service.rs

```diff
@@ -55,6 +52,13 @@ where
 
 #[derive(Debug)]
 pub struct ServiceBuilder<E> {
+    id: Uri,
+    service_endpoint: Url,
+    extra: E,
+}
+
+#[derive(Debug)]
+pub struct ServiceBuilderWithServiceType<E> {
```

Comment:
First, note that this is a builder, not the final struct. Second, note that the final struct is (de)serialized with this field renamed to "type"; `service_type` is used since `type` is a reserved Rust keyword.